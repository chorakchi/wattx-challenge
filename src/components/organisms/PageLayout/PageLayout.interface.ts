import { HTMLAttributes } from "react";
import { menuItems } from "../Menu/Menu.interface";

export interface PageLayoutProps
  extends HTMLAttributes<HTMLElement> {
  title?: string;
  description?: any;
  menuItems?: Array<menuItems>;
  onChangeRoute?: (path: string) => void;
  currentPath?: string;
}
