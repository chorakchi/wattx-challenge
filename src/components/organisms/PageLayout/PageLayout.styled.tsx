import styled from "styled-components";

export const StyledWrapper = styled.div`
padding: 5%;
margin: 0;
height: 100%;
width: 100%;
display: grid;
grid-column: 1;
grid-auto-flow: column;
grid-template-columns: max-content;
gap: 128px;

@media screen and (max-width: 1280px) {
    display: flex;
    gap: 18px;
    padding: 2%;
    padding-bottom: 120px;
}
}

.mainBlock{
    width: 100%;
    height: 100%;
    gap: 18px;
    @media screen and (max-width: 1280px) {
gap:0;

}
`;
