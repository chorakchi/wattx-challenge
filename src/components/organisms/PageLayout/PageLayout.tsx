import { StyledWrapper } from "./PageLayout.styled";
import { PageLayoutProps } from "./PageLayout.interface";
import { Stack, Typography, Menu } from "components";
export const PageLayout: React.FC<PageLayoutProps> = ({
  children,
  title,
  description,
  menuItems,
  onChangeRoute,
  currentPath,
  ...props
}: PageLayoutProps) => {
  return (
    <StyledWrapper {...props}>
      <Menu items={menuItems} onChangeRoute={onChangeRoute} currentPath={currentPath} />
      <Stack direction="column" className="mainBlock">
        <Typography variant="headline_xl">{title}</Typography>
        <Typography>{description}</Typography>
        <Stack direction="row" className="mainBlock">
          {children}
        </Stack>
      </Stack>
    </StyledWrapper>
  );
};
