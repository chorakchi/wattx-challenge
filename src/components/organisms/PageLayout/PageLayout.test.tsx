import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { PageLayout } from "./PageLayout";

jest.mock("components/atoms/SVG/SVG", () => ({
  SVG: "svg",
}));

describe("PageLayout", () => {
  it("should render the PageLayout", () => {
    const subject = shallow(
      <PageLayout
        menuItems={[
          {
            icon: "home",
            uri: "/",
          },
        ]}
      />
    );
    expect(subject).toHaveLength(1);
  });

  it("should render correctly and match to the Snapshot", () => {
    const subject = renderer
      .create(
        <PageLayout
          menuItems={[
            {
              icon: "home",
              uri: "/",
            },
          ]}
          onChangeRoute={() => {}}
          currentPath=""
        />
      )
      .toJSON();
    expect(subject).toMatchSnapshot();
  });
});
