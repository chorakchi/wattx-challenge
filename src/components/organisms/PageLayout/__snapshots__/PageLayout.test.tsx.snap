// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLayout should render correctly and match to the Snapshot 1`] = `
<div
  className="PageLayoutstyled__StyledWrapper-sc-1kudg7l-0 cijTrw"
>
  <div
    className="Boxstyled__BoxStyled-sc-12nhtob-0 bOPHwF Menustyled__StyledWrapper-sc-1vjf9c7-0 gkofae"
  >
    <div
      className="Menustyled__StyledItem-sc-1vjf9c7-1 dUQqyy"
      onClick={[Function]}
      selected={true}
    >
      <svg
        name="home"
        size="medium"
      />
    </div>
    <div
      className="Menustyled__StyledGap-sc-1vjf9c7-2 ddQrIZ"
    />
  </div>
  <div
    className="Stackstyled__StackStyled-sc-1adxq1u-0 bnrRyH mainBlock"
    direction="column"
  >
    <div
      className="Typographystyled__TypographyStyled-sc-i22hyo-0 jzzjid"
    />
    <div
      className="Typographystyled__TypographyStyled-sc-i22hyo-0 FBBmm"
    />
    <div
      className="Stackstyled__StackStyled-sc-1adxq1u-0 jWmaRX mainBlock"
      direction="row"
    />
  </div>
</div>
`;
